test/helpers/contract-test-helpers.js

```diff
@@ -25,12 +25,55 @@ async function mineBlock() {
   return await lastBlockNumber()
 }
 
+// This function is meant to be used along with the Hardhat forking feature
+// (https://hardhat.org/guides/mainnet-forking.html). It resets the fork state
+// to the given origin block. It is especially useful in system tests
+// environment which leverage mainnet forking feature. For example, it
+// can be used to set the environment to the same deterministic state, before
+// each test case.
+async function resetFork(blockNumber) {
+  await hre.network.provider.request({
+    method: "hardhat_reset",
+    params: [
+      {
+        forking: {
+          jsonRpcUrl: process.env.FORKING_URL,
+          blockNumber: blockNumber,
+        },
+      },
+    ],
+  })
+}
+
+async function impersonateAccount(accountAddress, purseSigner, amount) {
+  // Make the required call against Hardhat Runtime Environment.
+  await hre.network.provider.request({
+    method: "hardhat_impersonateAccount",
+    params: [accountAddress],
+  })
+
+  if (purseSigner) {
+    // Fund the account using a purse account in order to make transactions.
+    // In case the account represents a contract, keep in mind the contract must
+    // have a receive or fallback method to be funded successfully.
+    await purseSigner.sendTransaction({
+      to: accountAddress,
+      value: ethers.utils.parseEther(amount || "1"),
+    })
+  }
+
+  // Return the account's signer.
+  return await ethers.getSigner(accountAddress)
+}
+
```

Comment:
Non blocking but I think it would be worth looking if we can extract it to https://github.com/keep-network/hardhat-helpers and then import here.